src/app/clusters/time-synchronization-server/tests/BUILD.gn

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2025 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+import("//build_overrides/pigweed.gni")
+
+import("${chip_root}/build/chip/chip_test_suite.gni")
+
+chip_test_suite("tests") {
+  output_name = "libTestTimeSynchronizationCluster"
+
+  test_sources = [ "TestTimeSynchronizationCluster.cpp" ]
+
+  sources = []
+
+  cflags = [ "-Wconversion" ]
+
+  public_deps = [
+    "${chip_root}/src/app/clusters/testing",
+    "${chip_root}/src/app/clusters/time-synchronization-server",
+    "${chip_root}/src/app/server-cluster/testing:testing",
+    "${chip_root}/src/controller/data_model:data_model",
```

Comment:
This dependency should not be there ... tests depend on mock data model and code driven clusters should not depend on any data model at all.